test/widgets/autocomplete_test.dart

```diff
@@ -77,34 +89,59 @@ void main() {
       await tester.enterText(composeInputFinder, 'hello @user ');
       await tester.enterText(composeInputFinder, 'hello @user t');
       await tester.pumpAndSettle(); // async computation; options appear
+
       // "User Two" and "User Three" appear, but not "User One"
       check(tester.widgetList(find.text('User One'))).isEmpty();
+      final user1AvatarFinder =
+          findNetworkImage(store.tryResolveUrl('user1.png').toString());
+      check(tester.widgetList(user1AvatarFinder)).isEmpty();
+
       tester.widget(find.text('User Two'));
+      final user2AvatarFinder =
+          findNetworkImage(store.tryResolveUrl('user2.png').toString());
+      tester.widget(user2AvatarFinder);
```

Comment:
If we do include this new test logic as part of this same test case, though, it's important to keep the original test narrative readable. Part of keeping it readable is keeping it concise, without too many details in between each of the comments that are telling each step of the story.

Relatedly, this comment `// "User Two" and "User Three" appear, but not "User One"` is a heading visually attached to the three lines below it. When those become 14 lines including two blank lines, it becomes a lot harder to visually parse the comment as describing that whole group.

The basic strategy for keeping it concise is to abstract details into helpers. You already have these helpers `user1AvatarFinder`, etc., and so a first step would be to just pull those up into the setup part of this function (above the first narrative comment) so that this can read:
```dart
      // "User Two" and "User Three" appear, but not "User One"
      check(tester.widgetList(find.text('User One'))).isEmpty();
      check(tester.widgetList(user1AvatarFinder)).isEmpty();
      tester.widget(find.text('User Two'));
      tester.widget(user2AvatarFinder);
      tester.widget(find.text('User Three'));
      tester.widget(user3AvatarFinder);
```

That works, but I think we can do better by bringing it back down to one line per user. Try writing a helper so that this can look like:
```dart
      // "User Two" and "User Three" appear, but not "User One"
      checkUserShown(user1, expected: false);
      checkUserShown(user2, expected: true);
      checkUserShown(user3, expected: true);
```

Reply:
I was also thinking about the same helper method!! As there will be more bits added to it in the future, specifically with the user pronouns on the way soon!